Fails to build with only `BUILD_SINGLE` and `BUILD_DOUBLE` on Windows

Building OpenBLAS fails on Windows when enabling only single and double (`-DBUILD_SINGLE=ON -DBUILD_DOUBLE=ON`
```
[2868/5474] Building C object driver/others/CMakeFiles/driver_others.dir/blas_server_win32.c.obj
FAILED: driver/others/CMakeFiles/driver_others.dir/blas_server_win32.c.obj
D:\Programs\msys64_test\clang64\bin\clang.exe  -ID:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21 -ID:/dev/MINGW-packages/mingw-w64-openblas/src/build-CLANG64-32 -DHAVE_C11 -DMS_ABI -Wall -m64 -DSMALL_MATRIX_OPT -DDYNAMIC_ARCH -DSMP_SERVER -DMAX_CPU_NUMBER=64 -DMAX_PARALLEL_NUMBER=1 -DNO_AFFINITY -DVERSION="\"0.3.21\"" -DBUILD_SINGLE -DBUILD_DOUBLE -O3 -DNDEBUG -MD -MT driver/others/CMakeFiles/driver_others.dir/blas_server_win32.c.obj -MF driver\others\CMakeFiles\driver_others.dir\blas_server_win32.c.obj.d -o driver/others/CMakeFiles/driver_others.dir/blas_server_win32.c.obj -c D:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21/driver/others/blas_server_win32.c
D:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21/driver/others/blas_server_win32.c:298:40: error: use of undeclared identifier 'ZGEMM_P'
              sb = (void *)(((BLASLONG)sa + ((ZGEMM_P * ZGEMM_Q * 2 * sizeof(double)
                                              ^
D:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21/driver/others/blas_server_win32.c:298:50: error: use of undeclared identifier 'ZGEMM_Q'
              sb = (void *)(((BLASLONG)sa + ((ZGEMM_P * ZGEMM_Q * 2 * sizeof(double)
                                                        ^
D:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21/driver/others/blas_server_win32.c:301:40: error: use of undeclared identifier 'CGEMM_P'
              sb = (void *)(((BLASLONG)sa + ((CGEMM_P * CGEMM_Q * 2 * sizeof(float)
                                              ^
D:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21/driver/others/blas_server_win32.c:301:50: error: use of undeclared identifier 'CGEMM_Q'
              sb = (void *)(((BLASLONG)sa + ((CGEMM_P * CGEMM_Q * 2 * sizeof(float)
                                                        ^
D:/dev/MINGW-packages/mingw-w64-openblas/src/OpenBLAS-0.3.21/driver/others/blas_server_win32.c:567:30: warning: cast to 'void *' from smaller integer type 'long' [-Wint-to-void-pointer-cast]
                                     blas_thread_server, (void *)i,
                                                         ^~~~~~~~~
1 warning and 4 errors generated.
```

--- a/driver/others/blas_server_win32.c
+++ b/driver/others/blas_server_win32.c
@@ -278,12 +278,15 @@ static DWORD WINAPI blas_thread_server(void *arg){
 	  } else
 #endif
 	    if ((queue -> mode & BLAS_PREC) == BLAS_DOUBLE){
+#ifdef BUILD_DOUBLE
 	      sb = (void *)(((BLASLONG)sa + ((DGEMM_P * DGEMM_Q * sizeof(double)
 					  + GEMM_ALIGN) & ~GEMM_ALIGN)) + GEMM_OFFSET_B);
-
+#endif
 	    } else if ((queue -> mode & BLAS_PREC) == BLAS_SINGLE) {
+#ifdef BUILD_SINGLE
 	      sb = (void *)(((BLASLONG)sa + ((SGEMM_P * SGEMM_Q * sizeof(float)
 					  + GEMM_ALIGN) & ~GEMM_ALIGN)) + GEMM_OFFSET_B);
+#endif
 	    } else {
             /* Other types in future */
 	    }
@@ -295,11 +298,15 @@ static DWORD WINAPI blas_thread_server(void *arg){
 	  } else
 #endif
 	    if ((queue -> mode & BLAS_PREC) == BLAS_DOUBLE){
+#ifdef BUILD_COMPLEX16
 	      sb = (void *)(((BLASLONG)sa + ((ZGEMM_P * ZGEMM_Q * 2 * sizeof(double)
 					  + GEMM_ALIGN) & ~GEMM_ALIGN)) + GEMM_OFFSET_B);
+#endif
 	    } else if ((queue -> mode & BLAS_PREC) == BLAS_SINGLE) {
+#ifdef BUILD_COMPLEX
 	      sb = (void *)(((BLASLONG)sa + ((CGEMM_P * CGEMM_Q * 2 * sizeof(float)
 					  + GEMM_ALIGN) & ~GEMM_ALIGN)) + GEMM_OFFSET_B);
+#endif
 	    } else {
             /* Other types in future */
 	    }

